app/pages/project/stats/stats.cjsx

```diff
@@ -0,0 +1,213 @@
+        else
+          range.push(undefined)
+    else
+      range = [undefined, undefined]
+    workflowSelect = @workflowSelect()
+    <div>
+      {@props.type[0].toUpperCase() + @props.type.substring(1)}s per{' '}
+      <select value={@props.by} onChange={@handleGraphChange.bind this, @props.type}>
+        <option value="hour">hour</option>
+        <option value="day">day</option>
+        <option value="week">week</option>
+        <option value="month">month</option>
+      </select>
+      {workflowSelect}
+      <br />
+      <PromiseRenderer promise={@statCount(@props.by, @props.type)}>{(statData) =>
+        <Graph data={statData} by={@props.by} range={range} num={24} handleRangeChange={@handleRangeChange} />
+      }</PromiseRenderer>
+    </div>
+
+  handleGraphChange: (which, e) ->
+    @props.handleGraphChange(which, e)
+    
+  handleRangeChange: (range) ->
+    @props.handleRangeChange(@props.type, range)
+
+ETA = React.createClass
+  render: ->
+    if @props.data.length > 15
+      # don't count the last bin since the current day is not over yet
+      value = @props.data.slice(@props.data.length - 15, @props.data.length - 1)
+      days = 14
+    else
+      value = @props.data.slice(0, @props.data.length - 1)
+      days = @props.data.length - 1
+    rate = value.reduce (a,b) -> a + b
+    eta = Math.max(0, Math.ceil(days * (@props.totalCount - @props.currentCount) / rate))
+    <div>
+      <span className="progress-stats-label">ETC*</span> {"#{eta} days"}
+    </div>
+
+WorkflowProgress = React.createClass
+  statCount: ->
+    stats_url = "#{config.statHost}/counts/classification/day?workflow_id=#{@props.workflow.id}"
+    makeHTTPRequest 'GET', stats_url
+      .then (response) =>
+        results = JSON.parse response.text
+        bucket_data = results["events_over_time"]["buckets"]
+        data = bucket_data.map (stat_object) =>
+          stat_object.doc_count
+      .catch (response) ->
+        console?.error 'Failed to get the stats'
+        
+  render: ->
+    if @props.workflow.retirement.criteria == 'classification_count'
+      retirement = <div><span className="progress-stats-label">Retirement limit:</span> {@props.workflow.retirement.options.count.toLocaleString()}</div>
+    <div className="progress-element">
+      <div className="flex-wrapper">
+        <h3>{@props.workflow.display_name}</h3>
+        <div>
+          {retirement}
+        </div>
+        <div>
+          <span className="progress-stats-label">Images retired:</span> {@props.workflow.retired_set_member_subjects_count.toLocaleString()} / {@props.workflow.subjects_count.toLocaleString()}
+        </div>
+        <div>
+          <span className="progress-stats-label">Classifications:</span> {@props.workflow.classifications_count.toLocaleString()} / {(@props.workflow.subjects_count * @props.workflow.retirement.options.count).toLocaleString()}
+        </div>
+        <PromiseRenderer promise={@statCount()}>{(statData) =>
```

Comment:
We're trying to remove the `PromiseRenderer` component. See @brian-c's comments in the [PFE readme](https://github.com/zooniverse/Panoptes-Front-End/blob/master/README.md)

Better to store the stat data in component state and render when it updates.